internal/controllers/historicalv2.go

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/DIMO-Network/device-data-api/internal/services/elastic"
+	"github.com/DIMO-Network/shared/middleware/privilegetoken"
+	"github.com/gofiber/fiber/v2"
+)
+
+// GetHistoricalPermissionedV2 godoc
+// @Description  Get all historical data for a tokenID, within start and end range
+// @Tags         device-data
+// @Produce      json
+// @Success      200
+// @Param        tokenID  path   int64  true   "token id"
+// @Param        startTime     query  string  false  "startTime is an RFC3339 formatted date-time string. if empty two weeks back"
+// @Param        endTime       query  string  false  "endTime is an RFC3339 formatted date-time string. if empty today"
+// @Param        buckets       query  string  false  "number of data points to return, default 1000"
+// @Security     BearerAuth
+// @Router       /v2/vehicle/{tokenID}/history [get]
+func (d *DeviceDataController) GetHistoricalPermissionedV2(c *fiber.Ctx) error {
+	// get the startTime and endTime from the query parameters and convert them to time.Time
+	var err error
+	startTime, err := getTimeFromQuery(c, "startTime")
```

Comment:
Did you see [`QueryParser`](https://docs.gofiber.io/api/ctx/#queryparser)? Does that help at all?

Reply:
I saw it, but I would still need to convert the fields to `startTime` and `endTime` to `time.Time` objects and `buckes` to an int so I didn't think the abstraction had a huge benefit. I am not against it if we want to start using it consistently.

Reply:
I think that function would do the conversion for you. That's what I would be interested in.